### ISSUE TITLE
CRA Sass/scss build error: "resolve-url-loader: error processing CSS   Invalid mapping"

### ISSUE THREAD

**Issue by alexhmm** [tailwindlabs/tailwindcss#8806]
**What version of Tailwind CSS are you using?**

v3.1.5

**What build tool (or framework if it abstracts the build tool) are you using?**

Create React App

**What version of Node.js are you using?**

v16.14.2

**What browser are you using?**

Chrome, Firefox, Edge

**What operating system are you using?**

Windows 11

**Reproduction URL**

See attachment.

**Describe your issue**

Using tailwindcss with sass `@apply` with arbituary values at the end on the line causes following error when using `npm run build`:

Error: resolve-url-loader: error processing CSS
  Invalid mapping: {"generated":{"line":1,"column":22},"source":"file://C:\\...\react-tailwindcss-build-error\\src\\App.module.scss","original":{"column":null},"name":null}
  
This problem is also described here:
  
https://lakur.tech/2022/03/31/tailwind-scss-resolve-url-loader-error-processing-css-invalid-mapping/
  
The attachment shows a new setup project with a scss module file using the `@apply` rule.

**Steps to reproduce**
- Run `npm install` and `npm run build` in attached project

[react-tailwindcss-build-error.zip](https://github.com/tailwindlabs/tailwindcss/files/9071504/react-tailwindcss-build-error.zip)

**Comment by thecrypticace** [comment]
Hey, thanks for the report. This issue isn't on us so we unfortunately can't do anything to fix it. :(

This is an *old* postcss bug that has long since been fixed. However, CRA uses `resolve-url-loader` v4 which uses PostCSS v7. The problem is that the `.scss` module gets minified before being passed to the postcss version used by `resolve-url-loader` resulting in it being passed the following: `.width{@apply w-[200px]}`. The missing semicolon after an "at-rule" triggers a bug in older PostCSS builds that breaks generation of the source map: 
<img width="366" alt="Screen Shot 2022-07-08 at 09 57 28" src="https://user-images.githubusercontent.com/614993/178006337-d751f3a9-7b4c-446d-9ec6-e89097dfb877.png">

I believe the only reason it's being run through resolve-url-loader at all is because of the `.scss` module. If I move that CSS to either `index.css` or `App.css` it works fine. Also if you change the change the extension from `.scss` to `.css` it works fine.

I would suggest either:
1. Not using sass/scss to work around the parser bug if at all possible.
2. Not using `@apply` here since it just resolves to `width: 200px;`
3. Alternatively, adding a dummy CSS declaration after the `@apply` declaration like so:
```css
.width {
  @apply w-[200px];
  foo: bar;
}
```

Sorry we can't do more to work around this one :(

**Comment by alexhmm** [comment]
I am working on a big business project with scss modules. I just created this simple example to show the error.

Thx for the description and your time.

**Comment by SDIjeremy** [comment]
https://aronschueler.de/blog/2022/03/31/tailwind-scss-resolve-url-loader-error-processing-css-invalid-mapping/
TLDR: Move your tailwind one-off definitions like max-w-[500px] away from line endings, e.g.
```css
@apply mt-5 max-w-[500px];
```
should be
```css
@apply max-w-[500px] mt-5;
```

**Comment by aaroncrawford** [comment]
When using CRA + SCSS + Tailwind, I was getting this.  As @thecrypticace mentioned, it has to do with resolve-url-loader in CRA using version 4.  I created a sandbox and ejected CRA, then upgraded resolve-url-loader to version 5, and the builds started working.

So, without ejecting CRA, you can force CRA to use version 5 of resolve-url-loader by first installing resolve-url-loader as a dep:
`npm i resolve-url-loader@5.0.0`

Then in your package.json for NPM, force CRA to use v5 by adding this:
`"overrides": {
		"resolve-url-loader": "^5.0.0"
	}`
I believe in yarn `overrides` is called `resolutions` FYI.

I haven't noticed any issues so far by forcing CRA to use 5, but your mileage may vary.
